experimental/dds/tree2/src/feature-libraries/editable-tree/editableField.ts

```diff
@@ -184,6 +186,49 @@ export class FieldProxyTarget extends ProxyTarget<FieldAnchor> implements Editab
 		this.context.insertNodes(fieldPath, index, newContent);
 	}
 
+	public moveNodes(
+		sourceIndex: number,
+		count: number,
+		destinationIndex: number,
+		destinationField?: EditableField,
+	): void {
+		const sourceFieldPath = this.cursor.getFieldPath();
+		const destinationFieldKindIdentifier =
+			destinationField !== undefined
+				? destinationField.fieldSchema.kind.identifier
+				: this.fieldSchema.kind.identifier;
+
+		assert(
+			this.fieldSchema.kind.identifier === sequence.identifier &&
+				destinationFieldKindIdentifier === sequence.identifier,
+			"Both source and destination fields must be sequence fields.",
+		);
+
+		const destinationFieldProxy =
+			destinationField !== undefined
+				? (destinationField[proxyTargetSymbol] as ProxyTarget<Anchor | FieldAnchor>)
+				: this;
+
+		assert(
+			destinationFieldProxy === undefined ||
+				keyIsValidIndex(destinationIndex, this.length + 1),
```

Comment:
keyIsValidIndex isn't white what we want here since it tolerates strings as input, so has some extra overhead for that.

`assertValidIndex` is a better fit. Currently its just private to objectForest. Moving it to src/util/utils.ts, exporting it and making it take a more flexible input type for the array (`readonly unknown[]` if that works, `{length: number}` if you need to) would be better.

Reply:
moved assertValidIndex function to utils.ts